content/en/blog/2020/on-demand-egress-gateways/index.md

```diff
@@ -0,0 +1,540 @@
+---
+title: On Demand Egress Gateways
+subtitle: Deploy Egress Gateways independently in a seamless way
+description: Deploying Egress Gateways independently to fine-grained control of egress communication.
+publishdate: 2020-10-28
+attribution: Antonio Berben (At Deutsche Telekom - PAN-NET)
+keywords: [configuration,egress,gateway,external,service]
+target_release: 1.7.0
+---
+
+At [`Deutsche Telekom::PAN-NET`](https://pan-net.cloud/aboutus), we have embraced `Istio` as the umbrella to cover our services.
+
+Unfortunately, there are services which have not been migrated to `kubernetes` yet or, simply, cannot be. Therefore, those ones have to remain as external services.
+
+## Scenario
+
+If you are familiar with Istio, one of the offered ways to connect to external services is through [Egress Gateway](/docs/tasks/traffic-management/egress/egress-gateway/).
+
+However, if you want to satisfy the [Single-responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle), you will need to deploy multiple and individual (1..N) `Egress Gateways` as the picture shows.
```

Comment:
Is this really true? Can't you use a single egress gateway, just with multiple `Gateway` resources, to achieve this?

Reply:
Hi @frankbu , thank you for taking the time to read the article.
You can. It is true. And if the goal is to have single responsibility from the point of view of one egress for all egress communications, it makes sense.

But,

Making the story short: it is like monolith and microservices. Yes, you can do everything in one. But in some scenarios, it might not be very convenient, right?

Long story: Now, if the intention is to really apply single responsibility, and only one egress application per external service (or small meaningful group of services), then you have the two options described in the article: 1. Define them in the IstioOperator manifest (can become quite big). 2. Create your own set of egress services (what is in the hand-on).

As advantage of option 2, like the article explains, it is that, then, you have full control of the deployment object, and you can apply sidecar as your wish. One of the pictures shows that you would be able to add an OPA sidecar to integrate with OPA for AuthZ purposes. Therefore, your istio world and possibilities expands to infinite! :smile:

Let me know what you think. Thanks